torch/_inductor/codecache.py

```diff
@@ -496,7 +497,25 @@ def load(cls, source_code):
             lock_dir = get_lock_dir()
             lock = FileLock(os.path.join(lock_dir, key + ".lock"), timeout=LOCK_TIMEOUT)
             with lock:
-                output_path = input_path[:-3] + "so"
+                output_path = f"{input_path[:-3]}.so"
+
+                if config.aot_codegen:
+                    from .codegen.wrapper import CppWrapperCodeGen
+
+                    assert (
+                        cls.output_path is None
+                    ), "AOT compilation should only generate a single .so file"
+                    cls.output_path = output_path = os.path.join(
+                        os.getcwd(), f"{config.aot_codegen_output_prefix}.so"
+                    )
+                    output_header = os.path.join(
+                        os.getcwd(), f"{config.aot_codegen_output_prefix}.h"
+                    )
+                    with open(output_header, "w") as header_file:
+                        header_file.writelines("#include <torch/torch.h>\n\n")
+                        header_file.writelines(f"{CppWrapperCodeGen.decl_str};\n")
+                    log.info(f"AOT-Inductor compiles code into: {output_path}")
+
```

Comment:
I'd suggest moving this logic somewhere else.  Changing the behavior of `load()` to have side effects like this is confusing.